src/notify_service.hpp

```diff
@@ -13,6 +13,33 @@ namespace data_sync::notify
 
 namespace fs = std::filesystem;
 
+/**
+ * @brief The structure contains all retry-related details
+ *        if notification fails for a service
+ */
+struct Retry
```

Comment:
Why one more retry type when there is one already?

and why do we need this? I think we decided to use default attempts and interval, right? please use those directly - I feel its too much logic? any thoughts?